src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,195 @@
+
+	function handleAddAuthor(e, authorToBeAdded) {
+		e.preventDefault();
+		setCourseAuthorsList([
+			...courseAuthorsList,
+			{ id: authorToBeAdded.id, name: authorToBeAdded.name },
+		]);
+	}
+
+	function handleDeleteAuthor(e, authorToBeDeleted) {
+		e.preventDefault();
+		for (let author of courseAuthorsList)
+			if (author.id === authorToBeDeleted.id) {
+				setCourseAuthorsList(
+					courseAuthorsList.filter(
+						(author) => author.id !== authorToBeDeleted.id
+					)
+				);
+			}
+	}
+
+	function handleCreateAuthor(e) {
+		e.preventDefault();
+		if (authorToBeCreated.length < 2) {
+			alert('Author name must be at least 2 characters long.');
+			return;
+		}
+		setAuthorsList([
+			...authorsList,
+			{
+				id: uuid(),
+				name: authorToBeCreated,
+			},
+		]);
+		setAuthorToBeCreated('');
+	}
+
+	function handleCreateCourse(e) {
+		let creationDate = new Date();
+
+		e.preventDefault();
+		if (courseAuthorsList.length === 0) {
+			alert('Authors need to be added.');
+			return;
+		}
+		if (courseDuration === 0) {
+			alert('Course duration must be greater than 0.');
+			return;
+		}
+		if (courseDescription.length < 2) {
+			alert('Course description must be at least 2 characters long.');
+			return;
+		}
+		setCoursesList([
+			...coursesList,
+			{
+				id: uuid(),
+				title: courseTitle,
+				description: courseDescription,
+				creationDate: creationDate,
+				duration: courseDuration,
+				authors: courseAuthorsList.map((author) => author.id),
+			},
+		]);
+		setCreateCourseMode(false);
+	}
+
+	return (
+		<form onSubmit={handleCreateCourse}>
+			<div className='title-and-create'>
+				<Input
+					id='course-title-input'
+					className='course-title'
+					value={courseTitle}
+					onInput={(e) => setCourseTitle(e.target.value)}
+					placeholder='Enter title...'
+					labelClassName='course-title-input-label'
+					labelText='Title'
+					required
+				/>
+				<div className='create-course'>
+					<Button buttonText='Create course' className='create-course-button' />
+				</div>
+			</div>
+			<div className='description'>
+				<label htmlFor='course-description' className='description-label'>
+					Description
+				</label>
+				<textarea
+					id='course-description'
+					className='description-textarea'
+					placeholder='Enter description...'
+					value={courseDescription}
+					onInput={(e) => setCourseDescription(e.target.value)}
+					required
+				/>
+			</div>
+			<div className='course-info'>
+				<div className='create-author-and-duration'>
+					<h2>Add author</h2>
+					<Input
+						id='create-author-input'
+						className='create-author'
+						value={authorToBeCreated}
+						onInput={(e) => setAuthorToBeCreated(e.target.value)}
+						placeholder='Enter author name...'
+						labelClassName='create-author-input-label'
+						labelText='Author name'
+					/>
+					<Button
+						className='create-author-button'
+						buttonText='Create author'
+						onClick={(e) => handleCreateAuthor(e)}
```

Comment:
Please check this `onClick={handleCreateAuthor}`. This syntax works just the same but is shorter.
Have a look at the input here:
https://reactjs.org/docs/forms.html#controlled-components

Reply:
Understood. Changed :)